src/testdrive/src/main.rs

```diff
@@ -93,6 +97,7 @@ async fn main() {
         let _ = err.print_stderr(ci_output);
         process::exit(1);
     }
+    println!("testdrive completed successfully.");
 }
```

Comment:
Since `run` might print errors as it goes, I think the separation between `main` and `run` no longer makes sense. I would recommend merging the two functions. That way you won't need to construct an `Error::General` in `run` just to have it be printed here; you can just `eprintln!` stuff directly when `error_files.is_empty()`.